### ISSUE TITLE
Too many top-level files+dirs

### ISSUE THREAD

**Issue by JanCVanB** [roc-lang/roc#3966]
I think we, once again, have too many top-level files & directories. Here are screenshots of how zoomed-out my browser and terminal need to be to see them all (without column breaks, in the terminal's case):

![Screenshot from 2022-09-05 00-10-18](https://user-images.githubusercontent.com/4754114/188372076-a1cd6ef4-a0a3-4528-b6fb-07c88db00574.png)
![Screenshot from 2022-09-05 00-08-07](https://user-images.githubusercontent.com/4754114/188372079-72898cc6-0ab3-4c60-a16d-782872d1e8a4.png)

**Comment by JanCVanB** [comment]
I think this is both unwelcoming for newcomers and unwieldy for contributors. Does anyone else agree?

**Comment by JanCVanB** [comment]
How many directories is too many? Files? Should hidden vs. unhidden affect our self-imposed limits?

**Comment by brian-carroll** [comment]
I don't agree with this.

For newcomers there are several beginner-oriented markdown files among those top level files. And there are only a few directories that have clear meanings. The rest is all config files for tools, which, in my experience, every project has.

I would also question how many beginners in a language actually browse the compiler repo. Or even experienced engineers. How often have you done that in your usual language? Remember the stage we're at. Working in the compiler repo as a language user is not normal.

As a contributor I don't find it unwieldy at all. I actually think it's pretty compact. It would be an interesting challenge to find projects of similar size that don't have root directories like this. I randomly picked a few and they all look about the same to me.
https://github.com/rust-lang/rust
https://github.com/python/cpython
https://github.com/rails/rails

I guess different people have different experiences. What kind of projects are you comparing to?

**Comment by Anton-4** [comment]
I don't think it matters a lot for longtime contributors but it probably does for newcomers, it's hard to deny that showing lots of new information is overwhelming.
If we can reduce the list without disadvantages we definitely should.

My suggestions:
- Create a folder CONTRIBUTING with:
    + AUTHORS
    + BUILDING_FROM_SOURCE.md
    + CODE_OF_CONDUCT.md
    + CONTRIBUTING.md
- put FAQ.md, TUTORIAL.md and roc-for-elm-programmers.md in the getting_started folder

Earthfile and .earthignore can be deleted in the near future.

**Comment by folkertdev** [comment]
nesting such things makes them less discoverable. I think that is actually a bad thing. The repository should only be relevant for those wanting to contribute to roc (at least long term, that may be slightly different now) and we should optimize for that target audience, making the information that they would look for, easy to find (i.e. not nested). 

Consider other languages

- [rust](https://github.com/rust-lang/rust)
- [julia](https://github.com/JuliaLang/julia)
- [python](https://github.com/python/cpython)
- [ruby](https://github.com/ruby/ruby)
- [haskell](https://github.com/ghc/ghc)
- [zig](https://github.com/ziglang/zig)

all of these go "below the fold" with the list of files

**Comment by Anton-4** [comment]
Nesting can make things harder to find, but it can also guide people. It's a tricky balance :)

**Comment by JanCVanB** [comment]
Hooray for nuanced and contextual perspectives 😃 I'm now convinced that language/compiler repos necessitate above-average top-level breadth. Once the roc-lang org has more repos that are designed for newcomers & simplicity, that will also alleviate my "unwelcoming" concern.
